docs/template_plugin/tests/functional/op_reference/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace reference_tests;
+
+struct ExperimentalROIParams {
+    ExperimentalROIParams(const std::vector<Tensor>& experimental_detectron_roi_feature_inputs,
+                          const std::vector<Tensor>& expected_results,
+                          const std::string& test_case_name)
+        : inputs{experimental_detectron_roi_feature_inputs},
+          expected_results{expected_results},
+          test_case_name{test_case_name} {}
+
+    std::vector<Tensor> inputs;
+    std::vector<Tensor> expected_results;
+    std::string test_case_name;
+};
+
+class ReferenceExperimentalROILayerTest : public testing::TestWithParam<ExperimentalROIParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = create_function(params.inputs);
+        inputData.reserve(params.inputs.size());
+        refOutData.reserve(params.expected_results.size());
+        for (auto& input_tensor : params.inputs) {
+            inputData.push_back(input_tensor.data);
+        }
+        for (auto& expected_tensor : params.expected_results) {
```

Comment:
nitpick: in both cycles you can use `const auto&`

Reply:
Fixed.